org.jenetics.example/src/main/java/org/jenetics/example/image/ImageEvolutionExample.java

```diff
@@ -0,0 +1,299 @@
+            bar.add(picLabel);
+
+            painter = new ImagePainter(origImage.getWidth(), origImage.getHeight());
+            bar.add(painter);
+
+            add(bar, BorderLayout.CENTER);
+
+            JButton startButton = new JButton("Start");
+            startButton.setActionCommand("start");
+            add(startButton, BorderLayout.EAST);
+
+            bestFitLabel = new JLabel("Click on the start button...");
+            add(bestFitLabel, BorderLayout.SOUTH);
+
+            startButton.addActionListener(new ActionListener() {
+                @Override
+                public void actionPerformed(ActionEvent e) {
+                    if (isAlive()) {
+                        stopRequest();
+                        startButton.setText("Start");
+                    } else {
+                        startButton.setText("Stop");
+                        startEvolution();
+                    }
+                }
+            });
+
+            // initialize a new genetic algorithm
+            final Factory<Genotype<PolygonGene>> gtf = Genotype.of(new PolygonChromosome(POLYGON_COUNT, POLYGON_LENGTH));
+
+            engine = Engine
+                    .builder(ImageEvolutionExample::fitness, gtf)
+                    .populationSize(POPULATION_SIZE)
+                    .optimize(Optimize.MAXIMUM)
+                    .survivorsSelector(new TruncationSelector<>())
+                    .offspringSelector(new TournamentSelector<>(TOURNAMENT_ARITY))
+                    .alterers(
+                              new PolygonMutator<Double>(MUTATION_RATE, MUTATION_CHANGE),
+                              new UniformCrossover<>(0.5)
+                             )
+                    .build();
+        }
+
+        public boolean isAlive() {
+            return internalThread != null && internalThread.isAlive();
+        }
+
+        public void stopRequest() {
+            stopRequested = true;
+            internalThread.interrupt();
+            internalThread = null;
+        }
+
+        public void startEvolution() {
+            stopRequested = false;
+            Runnable r = new Runnable() {
+                @Override
+                public void run() {
+                    Iterator<EvolutionResult<PolygonGene, Double>> evolutionIterator = engine.iterator();
+                    while (!stopRequested && evolutionIterator.hasNext()) {
+                        EvolutionResult<PolygonGene, Double> nextEvolution = evolutionIterator.next();
+                        bestFit = (PolygonChromosome) nextEvolution.getBestPhenotype().getGenotype().getChromosome();
+                        bestFitLabel.setText(String.format("generation: %d - bestFit: %.4f",
+                                                           nextEvolution.getGeneration(), nextEvolution.getBestPhenotype().getFitness()));
+                        painter.repaint();
+                    }
+                }
+            };
```

Comment:
The working thread always shows the best solution of the current population. To show the best solution found so far, we must safe it:
```java
Runnable r = new Runnable() {
	final MinMax<EvolutionResult<PolygonGene, Double>> best = MinMax.of();
	@Override
	public void run() {
		final Iterator<EvolutionResult<PolygonGene, Double>>
			evolutionIterator = engine.iterator();

		while (!stopRequested) {
			best.accept(evolutionIterator.next());

			bestFit = (PolygonChromosome)best.getMax()
				.getBestPhenotype().getGenotype().getChromosome();

			bestFitLabel.setText(String.format(
				"generation: %d - bestFit: %.4f",
				best.getMax().getGeneration(),
				best.getMax().getBestPhenotype().getFitness()));

			painter.repaint();
		}
	}
};
```

The `evolutionIterator.hasNext()` check is not necessary, since it will always return `true`.